src/vl53l0x.c

```diff
@@ -0,0 +1,227 @@
+ *  if (Status == VL53L0X_ERROR_NONE) {
+ *      Status = VL53L0X_SetLimitCheckEnable(p_device,
+ *              VL53L0X_CHECKENABLE_RANGE_IGNORE_THRESHOLD, 1);
+ *  }
+ * //*/
+
+    // Set Values
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_SetLimitCheckValue(p_device,
+                                            VL53L0X_CHECKENABLE_SIGMA_FINAL_RANGE,
+                                            (FixPoint1616_t)(18 * 65536)); // Default: 18mm
+    }
+
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_SetLimitCheckValue(p_device,
+                                            VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
+                                            (FixPoint1616_t)(0.25 * 65536)); // Default: 0.25 Mcps
+    }
+
+    /*
+ *  if (Status == VL53L0X_ERROR_NONE) {
+ *      Status = VL53L0X_SetLimitCheckValue(p_device,
+ *              VL53L0X_CHECKENABLE_RANGE_IGNORE_THRESHOLD,
+ *              (FixPoint1616_t)(1.5*0.023*65536));
+ *  }
+ * //*/
+
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_SetMeasurementTimingBudgetMicroSeconds(p_device,
+                                                                50000); // Default: 33000us
+    }
+
+    // *
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_SetVcselPulsePeriod(p_device,
+                                             VL53L0X_VCSEL_PERIOD_PRE_RANGE, 18); // tem que ser entre 12 e 18
+    }
+
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_SetVcselPulsePeriod(p_device,
+                                             VL53L0X_VCSEL_PERIOD_FINAL_RANGE, 14); // tem q ser entre 8 e 14
+    }
+
+    // */
+
+    if (Status == VL53L0X_ERROR_NONE)
+    {
+        Status = VL53L0X_StartMeasurement(p_device);
+    }
+
+    return Status;
+}
+
+uint8_t vl53l0x_update_range(VL53L0X_Dev_t* p_device, VL53L0X_RangingMeasurementData_t* p_ranging_data,
+                            uint16_t* range, uint16_t max_range) {
+    uint8_t status = 0;
+    status = VL53L0X_GetRangingMeasurementData(p_device, p_ranging_data);
+
+    if (status != VL53L0X_ERROR_NONE) {
+        return status;
+    }
+
+    uint8_t range_status = p_ranging_data->RangeStatus; // TERMINAR A ANALISE DO STATUS
+
+    if (range_status == 0) { //TUDO OKAY
+        (*range) = (p_ranging_data->RangeMilliMeter) * FILTER_VALUE + (1 - FILTER_VALUE) * (*range);
+
+    } else if (range_status == 1) { //SIGMA FAIL
+        (*range) = (p_ranging_data->RangeMilliMeter) * 0.2 + (1 - 0.2) * (*range);
+    } else if (range_status == 4) { //PHASE FAIL
+        //Melhor não fazer nada?
+    } else if (range_status == 5) { //HARDWARE FAIL
+        (*range) = max_range;
+        //Desabilitar o sensor?
+    } else {
+        /*
+         * 2 - SIGNAL FAIL
+         * 3 - MIN RANGE FAIL
+         */
+        (*range) = (p_ranging_data->RangeMilliMeter) * 0.4 + (1 - 0.4) * (*range);
+    }
+
+
+    if ((*range) > max_range) {
```

Comment:
`*range = min(*range, max_range)`

Reply:
não sei se tenho essa função definida em algum lugar :p

Reply:
Defina asuhaush

Reply:
![image](https://user-images.githubusercontent.com/39196309/65980774-31776100-e44e-11e9-91eb-13049070bf36.png)

Reply:
Pode deixar no vl53l0x.h mesmo, ou talvez só no .c se nao for usar em outro lugar

Reply:
oki doki